opensearch-project/OpenSearch-Dashboards
# [Multiple Datasource] Add data source aggregated view to show all compatible data sources or only show used data sources

### Description
This change is based on requirement to show readonly view for all compatible connected data sources in the navigation bar, and also is capable of showing only the data sources passed by the plugins 

### Issues Resolved


## Screenshot

https://github.com/opensearch-project/OpenSearch-Dashboards/assets/17714150/226c0348-d83e-4264-bc88-808f067a5ea3


## Testing the changes
The following were performed in the recording above:
1. enable data source plugin
2. change prop to not display all compatible data sources and pass in active data source ids should show the info icon and clicking on it, will show drop down with title called `SELECTED DATA SOURCES`, the button should show the number of active/used data sources
3. change prop to display all compatible data sources and remove active data source ids should show the info icon and clicking on it, will show drop down with title called `DATA SOURCES` followed by number of compatible data sources and display them.

### Check List
